### PR TITLE
TS-1585 add test coverage for auth utils

### PR DESCRIPTION
**Pull request by LBHTKarki**
## WHAT
Add test coverage for googleAuth and related functions.

Files covered:
1. lib/utils/googleAuth.spec.ts
2. lib/utils/requestAuth.spec.ts
3. lib/utils/users.spec.ts

## WHY
Currently we don't have any test coverage for the functions involved in authentication. In order to develop the application further we need to add appropriate test coverage for the functions and components we modify. 

This adds baseline test coverage for the functions that are involved in authentication.

## HOW
Please note these tests are not opinionated about the current implementation, but cover the functionality as is. 

`requestAuth.ts` tests are lacking some mock setups due to the module structure which makes it difficult. Since this initial update is done without any refactoring or changes to the implementation this function is tested mainly against real implementatations.

jest and lint configs have been updated to ignore the new `testUtils` folder. Also some linting overrides have been added to test helpers to overcome some import issues that we haven't been able resolve in any other way yet.

### CODE REVIEW

File: testUtils/userHelper.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  Admin,
+  ReadOnly,
+}
+
+interface UserClaims {
+  hasAdminPermissions: boolean;
+  hasManagerPermissions: boolean;
+  hasOfficerPermissions: boolean;
+  hasReadOnlyPermissions: boolean;
+}
```
Comment (Duslerke) [inline testUtils/userHelper.ts:29]: This looks like a copy of `type Permissions` in the google auth file.

File: testUtils/userHelper.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    }
+    default:
+      return {
+        hasAdminPermissions: false,
+        hasManagerPermissions: false,
+        hasOfficerPermissions: false,
+        hasReadOnlyPermissions: false,
+      };
+  }
+};
```
Comment (Duslerke) [inline testUtils/userHelper.ts:92]: Could probably reduce its size a lot:

```ts
export const getClaimsByRole = (role?: UserRole): UserClaims => {
  const userPerms = {
        hasAdminPermissions: false,
        hasManagerPermissions: false,
        hasOfficerPermissions: false,
        hasReadOnlyPermissions: false,
      };

   switch (role) {
    case UserRole.ReadOnly: {
      userPerms.hasReadOnlyPermissions = true;
      break;
    }
    case UserRole.Officer: {
      userPerms.hasOfficerPermissions: true;
      break;
    }
    case UserRole.Manager: {
      userPerms.hasManagerPermissions = true;
      break;
    }
    case UserRole.Admin: {
      userPerms.hasAdminPermissions = true;
      break;
    }
    default:
      break; 
  }
  return userPerms;
};
```

File: lib/utils/googleAuth.spec.ts [hunk 3]
```diff
@@ -0,0 +1,483 @@
...
+      ];
+
+      const user = generateJWTTokenTestData(groupClaims, issuedAt);
+
+      const expectedPermissions: Permissions = {
+        hasAdminPermissions: true,
+        hasManagerPermissions: true,
+        hasOfficerPermissions: true,
+        hasReadOnlyPermissions: true,
+      };
```
Comment (Duslerke) [inline lib/utils/googleAuth.spec.ts:300]: Can the user have more than 1 set of permissions?

Comment (LBHTKarki) [reply lib/utils/googleAuth.spec.ts:300]: Technicaly they could be in multiple groups, but of course that's not how the permissions are setup in practise. However this particular test is just for checking that the claims are set correctly based on the provided groups.

File: testUtils/userHelper.ts (hunk 2 above)
Comment (LBHTKarki) [reply testUtils/userHelper.ts]: A lot tidier that way 👍 Updated in  [#0213be2](https://github.com/LBHackney-IT/lbh-housing-register/pull/395/commits/0213be25f6dcb0d99b84e50bf04aa55b15ca655a)

**Review by Duslerke — APPROVED** [review]

**Pull request merged**